### PR TITLE
CRS-127 psi example 16

### PR DESCRIPTION
**Pull request by joelstobart-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt [hunk 1]
```diff
@@ -62,6 +63,15 @@ class SentencesExtractionService {
     return sentenceStream.allMatch { sentence -> sentence.concurrentSentences.isEmpty() }
   }
 
+  fun allSentencesContainType(sentences: MutableList<Sentence>, sentenceType: SentenceType): Boolean {
+    for (sentence in sentences) {
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt:67]: We could modify this `for loop` to be more functional; e.g. something like this
`return sentences.any { sentence -> sentences.any { innerSentence -> doSentencesOverlap(sentence, innerSentence)} }`

**Review by yasinmustafa — DISMISSED** [review]
LGTM - one comment

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt [hunk 2]
```diff
@@ -64,12 +64,7 @@ class SentencesExtractionService {
   }
 
   fun allSentencesContainType(sentences: MutableList<Sentence>, sentenceType: SentenceType): Boolean {
-    for (sentence in sentences) {
-      if (!sentence.sentenceTypes.contains(sentenceType)) {
-        return false
-      }
-    }
-    return true
+    return sentences.stream().allMatch { sentence -> sentence.sentenceTypes.contains(sentenceType) }
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt:67]: Replaced with a stream based approach @yasinmustafa  :)

Comment (yasinmustafa) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt:67]: In kotlin you dont need to explicitly convert to a stream..s o the above could simply be (`it` is a keyword and refers to the current element in the list without having to explicitly name it):
`return sentences.all { it.sentenceTypes.contains(sentenceType) }`

**Review by yasinmustafa — APPROVED** [review]
Approved - with another minor comment, happy to defer

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt (hunk 2 above)
Comment (joelstobart-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt]: Nice - good learning experience :)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt [hunk 3]
```diff
@@ -64,7 +64,7 @@ class SentencesExtractionService {
   }
 
   fun allSentencesContainType(sentences: MutableList<Sentence>, sentenceType: SentenceType): Boolean {
-    return sentences.stream().allMatch { sentence -> sentence.sentenceTypes.contains(sentenceType) }
+    return sentences.all { it.sentenceTypes.contains(sentenceType) }
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt:67]: @yasinmustafa  Thank you for that - this works

**Review by yasinmustafa — APPROVED** [review]
LGTM

**Pull request merged**